Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp

```diff
@@ -3748,6 +3748,47 @@ JS_DEFINE_NATIVE_FUNCTION(@class_name@::@attribute.getter_callback@)
     if (content_attribute_value.has_value())
         retval = MUST(Infra::convert_to_scalar_value_string(*content_attribute_value));
 )~~~");
+            }
+            // If a reflected IDL attribute has the type T?, where T is either Element
+            // FIXME: or an interface that inherits from Element,
+            // then with attr being the reflected content attribute name:
+            else if (attribute.type->is_nullable() && attribute.type->name() == "Element") {
+                // The getter steps are to return the result of running this's get the attr-associated element.
+                attribute_generator.append(R"~~~(
+    auto retval = GC::Ptr<Element> {};                
+)~~~");
+
+                // 1. Let element be the result of running reflectedTarget's get the element.
+                // 2. Let contentAttributeValue be the result of running reflectedTarget's get the content attribute.
+                attribute_generator.append(R"~~~(
+    auto contentAttributeValue = impl->attribute(HTML::AttributeNames::@attribute.reflect_name@);
+)~~~");
+                // 3. If reflectedTarget's explicitly set attr-element is not null:
+                //      1. If reflectedTarget's explicitly set attr-element is a descendant of any of element's shadow-including ancestors, then return reflectedTarget's explicitly set attr-element.
+                //      2. Return null.
+                attribute_generator.append(R"~~~(
+    auto const explicitly_set_attr = TRY(throw_dom_exception_if_needed(vm, [&] { return impl->get_@attribute.cpp_name@(); }));
+    if (explicitly_set_attr) {
+        if (impl->shadow_including_root().is_equal_node(&explicitly_set_attr->shadow_including_root())) {
```

Comment:
This should be equivalent to `&impl->shadow_including_root() == &explicitly_set_attr->shadow_including_root()`